tests/modes/apps/extensions/test_extension_remote.py

```diff
@@ -186,3 +190,40 @@ def test_no_host_no_file_protocol_raises_assertion_error(self, mock_which: Magic
         with pytest.raises(AssertionError):
             # This creates a URL with no host but protocol is https
             parse_extension_url("https:///user/repo")
+
+
+class TestGenerateExtensionId:
+    def test_with_host_and_path(self) -> None:
+        assert generate_extension_id("github.com", "user/repo") == "com.github.user.repo"
+        assert generate_extension_id("example.co.uk", "user/repo") == "uk.co.example.user.repo"
+        assert generate_extension_id("gitlab.com", "group/subgroup/repo") == "com.gitlab.group.subgroup.repo"
```

Comment:
Do URLs like this exist? If they do our parser will break anyway, so I think the result should be that the parser should expect "group" to be the user id and "subgroup" to be the repo name, and "repo" to be ignored. Basically the same as the parser already treats "https://gitlab.com/user/repo/issues" by ignoring "issues"

(No need to address this feedback as #1504 should fix this)